tests/devices/unit_tests/areadetector/test_andor2.py

```diff
@@ -0,0 +1,151 @@
+from collections import defaultdict
+from pathlib import Path
+from unittest.mock import Mock, patch
+
+import pytest
+from bluesky.plans import count
+from bluesky.run_engine import RunEngine
+from ophyd_async.core import (
+    DetectorTrigger,
+    DeviceCollector,
+    FilenameProvider,
+    StaticFilenameProvider,
+    StaticPathProvider,
+    TriggerInfo,
+    assert_emitted,
+    callback_on_mock_put,
+    set_mock_value,
+)
+from ophyd_async.epics.adcore._core_io import DetectorState
+
+from dodal.devices.areadetector import Andor2
+from dodal.devices.areadetector.andor2_epics import (
+    Andor2DriverIO,
+    Andor2TriggerMode,
+    ImageMode,
+)
+from dodal.devices.areadetector.andor2_epics.andor2_controller import Andor2Controller
+
+
+@pytest.fixture
+def static_filename_provider():
+    return StaticFilenameProvider("ophyd_async_tests")
+
+
+@pytest.fixture
+def static_path_provider_factory(tmp_path: Path):
+    def create_static_dir_provider_given_fp(fp: FilenameProvider):
+        return StaticPathProvider(fp, tmp_path)
+
+    return create_static_dir_provider_given_fp
+
+
+@pytest.fixture
+def static_path_provider(
+    static_path_provider_factory,
+    static_filename_provider: FilenameProvider,
+):
+    return static_path_provider_factory(static_filename_provider)
+
+
+@pytest.fixture
+async def Andor() -> Andor2Controller:
+    async with DeviceCollector(mock=True):
+        drv = Andor2DriverIO("DRIVER:")
+        controller = Andor2Controller(drv)
+
+    return controller
+
+
+async def test_Andor_controller(RE, Andor: Andor2Controller):
```

Comment:
Should: Capitilisation is normally reserved for class names in Python so `Andor` looks like you're passing the class around, not the specific instance. I think it should be `andor`